src/components/eip4626/user/VaultDeposit.tsx

```diff
@@ -75,93 +72,139 @@ export function VaultDeposit({ vaultAddress }: VaultInfoProps) {
     });
   };
 
-  const { data: previewDepositData } = useReadHederaVaultPreviewDeposit(
-    vaultAddress,
-    form.values.amount,
+  const maxAmount = useMemo(
+    () => Number(formatBalance(vaultAssetUserBalance)),
+    [vaultAssetUserBalance],
   );
 
-  const previewDepositDataFormatted = formatBalance(
-    previewDepositData?.toString(),
-  );
+  const handleUpdateAmountWithPercent = (percentage: number | string) => {
+    const calculatedAmount =
+      typeof percentage === "number"
+        ? (maxAmount * percentage) / 100
+        : maxAmount;
+
+    if (calculatedAmount) {
+      depositForm.setFieldValue("amount", calculatedAmount);
+
+      if (calculatedAmount <= maxAmount) {
+        setMaxAmountError(false);
+      } else {
+        setMaxAmountError(true);
+      }
+    }
+  };
+
+  const [maxAmountError, setMaxAmountError] = useState(false);
 
   return (
     <>
-      <Heading size={"sm"}>Deposit asset into vault</Heading>
-
-      <form onSubmit={form.handleSubmit}>
-        <VStack gap={2} alignItems="flex-start">
-          <FormControl isRequired>
-            <FormLabel>Amount of asset to deposit</FormLabel>
-            <NumberInput
-              name="amount"
-              variant="outline"
-              value={form.values.amount}
-              clampValueOnBlur={false}
-              onChange={(val) => form.setFieldValue("amount", val)}
+      <form onSubmit={depositForm.handleSubmit}>
+        <Flex direction="column" gap="4">
+          <Flex
+            direction="row"
+            width="100%"
+            pt="2"
+            pb="2"
+            justifyContent="space-between"
+          >
+            <Text fontWeight="800" fontSize={14}>
+              Deposit Amount{": "}
+              {depositForm.values?.amount}
+            </Text>
+            <Text fontWeight="600" fontSize={14}>
+              Max: {maxAmount}
+            </Text>
+          </Flex>
+          <Flex width="100%" justifyContent="space-between" gap="1">
+            <Input
+              value={depositForm.values?.amount}
+              onChange={(e) => {
+                const calculatedAmount = parseFloat(e.target.value);
+
+                if (calculatedAmount || calculatedAmount === 0) {
+                  depositForm.setValues(() => ({
```

Comment:
why not .setFieldValue() ?
also why pass the function, i think it can accept the value right away?